### PR TITLE
internal: get pending and queued transaction by address

### PR DESCRIPTION
**Pull request by markya0616**
We can call Content() to get `all` pending and queued transactions, but the response is really large... 
This pull request implements an API to filter the pending/queued transactions by an address.

### CODE REVIEW

**Review by karalabe — COMMENTED** [review]
This seems very wasteful to retrieve all that data just to throw away most of it. I think if you wanted to get thins functionality in, we should add a `PendingFrom` and `QueuedFrom` into the txpool where we can retrieve only the data we need.

**Comment by karalabe** [comment]
Or to keep it consistent with te current API, `ContentFrom`

**Comment by markya0616** [comment]
@karalabe Thanks for your review. The PR is updated.

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -141,6 +141,36 @@ func (s *PublicTxPoolAPI) Content() map[string]map[string]map[string]*RPCTransac
...
+	curHeader := s.b.CurrentHeader()
+	// Build the pending transactions
+	txs, ok := pending[addr]
+	if ok {
+		dump := make(map[string]*RPCTransaction)
+		for _, tx := range txs {
+			dump[fmt.Sprintf("%d", tx.Nonce())] = newRPCPendingTransaction(tx, curHeader, s.b.ChainConfig())
+		}
+		content["pending"] = dump
+	}
```
Comment (holiman) [inline internal/ethapi/api.go:160]: Nitpick:
```suggestion
	if txs, ok := pending[addr]; ok {
		dump := make(map[string]*RPCTransaction)
		for _, tx := range txs {
			dump[fmt.Sprintf("%d", tx.Nonce())] = newRPCPendingTransaction(tx, curHeader, s.b.ChainConfig())
		}
		content["pending"] = dump
	}
```
And the same below

File: internal/ethapi/api.go [hunk 2]
```diff
@@ -141,6 +141,36 @@ func (s *PublicTxPoolAPI) Content() map[string]map[string]map[string]*RPCTransac
 	return content
 }
 
+// ContentFrom returns the transactions contained within the transaction pool.
+func (s *PublicTxPoolAPI) ContentFrom(addr common.Address) map[string]map[string]*RPCTransaction {
+	content := map[string]map[string]*RPCTransaction{
+		"pending": make(map[string]*RPCTransaction),
+		"queued":  make(map[string]*RPCTransaction),
```
Comment (holiman) [inline internal/ethapi/api.go:148]: You re-`make` these later on, so these one just get overwritten

**Comment by holiman** [comment]
Looking at it a bit more closely, I think this is what Peter meant. When you call TxpoolContent, it goes down into this method: 
```
func (pool *TxPool) Content() (map[common.Address]types.Transactions, map[common.Address]types.Transactions) {
	pool.mu.Lock()
	defer pool.mu.Unlock()

	pending := make(map[common.Address]types.Transactions)
	for addr, list := range pool.pending {
		pending[addr] = list.Flatten()
	}
	queued := make(map[common.Address]types.Transactions)
	for addr, list := range pool.queue {
		queued[addr] = list.Flatten()
	}
	return pending, queued
}

```
You could instead define a new method which doesn't even bother with all the addresses that aren't what you're looking for, and save a lot of work.

**Comment by markya0616** [comment]
@holiman, thanks for your review. I refined my PR based on your comments. :)

**Review by holiman — APPROVED** [review]
Looks good, thanks!
(caveat: I haven't tested it yet)

**Comment by markya0616** [comment]
@karalabe Do you have any thoughts about this PR?

**Comment by markya0616** [comment]
@karalabe Do you have any thoughts about this PR?

**Review by karalabe — APPROVED** [review]
LGTM

**Pull request merged**